Bump JDK source and target version

Travis CI currently fails with
```
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] Source option 5 is no longer supported. Use 6 or later.
[ERROR] Target option 1.5 is no longer supported. Use 1.6 or later.
[INFO] 2 errors 
```
E.g. https://travis-ci.org/github/biojava/biojava-legacy/builds/698663289

--- a/pom.xml
+++ b/pom.xml
@@ -57,7 +57,7 @@
     </developer>
   </developers>
   <properties>
-    <jdk.version>1.5</jdk.version>
+    <jdk.version>1.8</jdk.version>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <project.build.targetEncoding>UTF-8</project.build.targetEncoding>
     <maxmem>512m</maxmem>

